### ISSUE TITLE
SuperSetUserPassword pattern doesn't match description

### ISSUE THREAD

**Issue by asafh-lb** [aws-quickstart/quickstart-apache-superset#30]
Valid passwords according to the description get rejected by the pattern.

**Comment by yike5460** [comment]
Hi, will you please provide your input password? The regex to validate password pattern is like ^(?=.*[a-z])(?=.*[A-Z])(?=.*[@$!%*?&\\-_])[A-Za-z\\d@$!%*?&\\-_]{8,}$

**Comment by asafh-lb** [comment]
Hi,
I don't recall what did my original attempt looked like but for now, I see
a couple of ways the pattern is different from the description "Minimum of
8 characters, including 1 uppercase letter, 1 lowercase letter, 1 number,
and 1 symbol (excluding / @ \" ').":

   1. More strict regarding the special characters - (@$!%*?&-_) only and
   not things like #~<>+ etc
   2. Does not actually require a number ("aaaaaaA!" works)

From looking at the pattern now, I can only see it being more strict than
described in regards to the special character set.
I'm not sure what was the reasoning behind blocking some of the special
characters, but if that was mostly a way to make the regex simpler I can
suggest:

> ^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[\W_]).{8,}$
>
To add the requirement of the number, and to make the special character
more inclusive (any non-alphanumeric character)


Thanks

**Comment by yike5460** [comment]
Thanks for your suggestion, we may consider to rules below:

- password must contain 1 number (0-9)
- password must contain 1 uppercase letters
- password must contain 1 lowercase letters
- password must contain 1 non-alpha numeric number
- password is 8-16 characters with no space

and regex is like ^(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[^\w\d\s:])([^\s]){8,16}$

**Comment by yike5460** [comment]
new password regex update in official solution, kindly let me know your further feedback

**Comment by asafh-lb** [comment]
The markdown formatting altered your regex. What I assume was there:
`^(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[^\w\d\s:])([^\s]){8,16}$`

Looks pretty good with two caveats. For negated character set representing special characters, both underscore and colon are not considered (colon explicitly there, underscore due to \w). They would however be accepted in the general password.
I'm not sure what is the reasoning but if there's none I'd suggest to consider them as special characters and otherwise to state so in the rules' description.

Thank you for help.

**Comment by yike5460** [comment]
underscore and colon are removed from non-alpha numeric number list with regex ?=.*[^\w\d\s:], so password 'Aa12345@', 'Aa12345*' are accepted, while 'Aa12345_' and 'Aa12345:' not

**Comment by asafh-lb** [comment]
Right, what I meant to say that if those are not considered special characters by design I would note that in the description. "1 non-alpha numeric number excluding colon and underscore"
